skyportal/handlers/api/candidate.py

```diff
@@ -173,18 +172,20 @@ def get(self, obj_id=None):
                     joinedload(Candidate.obj).joinedload(Obj.thumbnails),
                     joinedload(Candidate.obj)
                     .joinedload(Obj.photometry)
-                    .joinedload(Photometry.instrument)
-                    .joinedload(Instrument.telescope),
+                    .joinedload(Photometry.instrument),
                 ],
             )
             if c is None:
                 return self.error("Invalid ID")
             candidate_info = c.to_dict()
             candidate_info["comments"] = sorted(
-                c.get_comments_owned_by(self.current_user),
-                key=lambda x: x.created_at,
+                [cmt.to_dict() for cmt in c.get_comments_owned_by(self.current_user)],
+                key=lambda x: x["created_at"],
                 reverse=True,
             )
+            for comment in candidate_info["comments"]:
+                comment["author"] = comment["author"].to_dict()
```

Comment:
Note this would have already happened but during the call to `success` -- this just converts to dict earlier rather than later so that user prefs can be deleted